app/controlplane/pkg/biz/group.go

```diff
@@ -853,3 +866,46 @@ func (uc *GroupUseCase) ValidateGroupIdentifier(ctx context.Context, orgID uuid.
 
 	return group.ID, nil
 }
+
+// ListProjectsByGroup retrieves a list of projects that a group is a member of with pagination.
+func (uc *GroupUseCase) ListProjectsByGroup(ctx context.Context, orgID uuid.UUID, opts *ListProjectsByGroupOpts, paginationOpts *pagination.OffsetPaginationOpts) ([]*GroupProjectInfo, int, error) {
+	if opts == nil {
+		return nil, 0, NewErrValidationStr("options cannot be nil")
+	}
+
+	if orgID == uuid.Nil {
+		return nil, 0, NewErrValidationStr("organization ID cannot be empty")
+	}
+
+	resolvedGroupID, err := uc.ValidateGroupIdentifier(ctx, orgID, opts.ID, opts.Name)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// Check the group exists
+	existingGroup, err := uc.groupRepo.FindByOrgAndID(ctx, orgID, resolvedGroupID)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to find group: %w", err)
+	}
+
+	if existingGroup == nil {
+		return nil, 0, NewErrNotFound("group")
+	}
+
+	pgOpts := pagination.NewDefaultOffsetPaginationOpts()
+	if paginationOpts != nil {
+		pgOpts = paginationOpts
+	}
+
+	var requesterMemberships []*Membership
+	// If requesterID is provided, list all memberships for the requester
+	if opts.RequesterID != nil {
```

Comment:
no need of this. get `visibleProjects` from the service and use it to create the filter.